packages/core-ui/lib/buttons/index.tsx

```diff
@@ -0,0 +1,111 @@
+/* @license Copyright 2023 @paritytech/polkadot-cloud authors & contributors
+SPDX-License-Identifier: Apache-2.0 */
+import {
+  ButtonCommonProps,
+  ButtonIconProps,
+  ComponentBaseWithClassName,
+} from "../types";
+import { ButtonPrimary, ButtonPrimaryProps } from "./ButtonPrimary";
+import {
+  ButtonPrimaryInvert,
+  ButtonPrimaryInvertProps,
+} from "./ButtonPrimaryInvert";
+import { ButtonSecondary, ButtonSecondaryProps } from "./ButtonSecondary";
+import { ButtonTertiary, ButtonTertiaryProps } from "./ButtonTertiary";
+import { ButtonMono, ButtonMonoProps } from "./ButtonMono";
+import { ButtonMonoInvert } from "./ButtonMonoInvert";
+import {
+  ButtonSubmitInvert,
+  ButtonSubmitInvertProps,
+} from "./ButtonSubmitInvert";
+import { ButtonText } from "./ButtonText";
+import { ButtonSubmit, ButtonSubmitProps } from "./ButtonSubmit";
+import { ButtonHelp, ButtonHelpProps } from "./ButtonHelp";
+import { ButtonTab, ButtonTabProps } from "./ButtonTab";
+import { ButtonOption, ButtonOptionProps } from "./ButtonOption";
+
+export type ButtonProps = ComponentBaseWithClassName &
```

Comment:
Perhaps just create a giant type:

```
export type ButtonType =
  | "help"
  | "mono"
  | "monoInvert"
  | "option"
  | "primary"
  | "primaryInvert"
  | "secondary"
  | "submit"
  | "submitInvert"
  | "tab"
  | "tertiary"
  | "text";

export type ButtonProps = ComponentBaseWithClassName & {
  // the type of button
  type: ButtonType;
} & (
    | ButtonHelpProps
    | ButtonMonoProps
    | ButtonMonoProps
    | ButtonOptionProps
    | ButtonPrimaryInvertProps
    | ButtonSecondaryProps
    | ButtonSubmitProps
    | ButtonSubmitInvertProps
    | ButtonTabProps
    | ButtonTertiaryProps
    | ButtonMonoProps
    | ButtonPrimaryProps
  );

```

Reply:
`ButtonType` exists in `packages/core-ui/lib/types.ts`. The `ButtonProps` also in a different format in order to reduce unnecessary repetitions;